cachi2/core/utils.py

```diff
@@ -74,3 +77,28 @@ def load_json_stream(s: str) -> Iterator:
     while match := non_whitespace.search(s, i):
         obj, i = decoder.raw_decode(s, match.start())
         yield obj
+
+
+def copy_directory(origin: Path, destination: Path, reflink_if_supported: bool = True) -> Path:
+    """
+    Recursively copy directory to another path.
+
+    Use reflinks by default if the file system supports it.
+
+    :param reflink_if_supported: if set to false, reflinks won't be used regardless of
```

Comment:
Since I bumped into a `ReflinkImpossibleError`, I figured there could be more corner cases in which we default to reflinks, only to make the request unprocessable. So the reasoning is that the user can, if he wants to, force Cachi2 to not use reflinks, although the default is using them if available.

We could have an automatic error fallback (default to reflink -> reflink copy fails -> try to use a normal copy), it is only more complex.

Reply:
Changed the source code to fallback to `shutil.copy2` in case a `ReflinkImpossibleError` happens.